snapcraft/internal/project_loader/_extensions/_utils.py

```diff
@@ -98,12 +98,9 @@ def find_extension(extension_name: str) -> Type[Extension]:
     except ImportError:
         raise errors.ExtensionNotFoundError(extension_name)
 
-    # This may throw an AttributeError, but that would be programmer error of whoever
-    # is hacking on extensions.
-    extension_class_name = "{}Extension".format(
-        extension_name.capitalize().replace("-", "_")
-    )
-    return getattr(extension_module, extension_class_name)
+    # Extensions have an attribute with the name of the extension
+    # class.
+    return getattr(extension_module, "ExtensionImpl")
```

Comment:
Seems a little odd to require them all to share the same name, but it works and simplifies the naming rule.

Reply:
As a follow up to your point on IRC.  I think the naming scheme is fine.  As implemented, it's already a requirement on the module name, with the added tweak to capitalize the class name.

I kinda liked how the plugin system found it by class type, but that's overkill and has some potential issues of its own.

I do often use class names to navigate source at times, so I'd lean to towards the naming format unless there's something technically wrong with it.  However, as it's largely a personal preference, I defer to your judgement.   :)